### PR TITLE
Added merge functions, comments, and export as PNG button

### PR DESCRIPTION
**Pull request by alwin48**
Screenshot (69)

### CODE REVIEW

File: src/graph.js [hunk 1]
```diff
@@ -9,25 +15,35 @@ class Graph extends React.Component {
...
 
     };
 
+    //Function to add branch and adds branch name to branches array in state
     addBranch = () => {
+        if (this.state.branches
+            .map((b) => b.name)
+            .includes(this.state.branchName)
+        )
+                  return;
```
Comment (anandbaburajan) [inline src/graph.js]: Style nitpick: can you maybe improve the formatting a bit? The `return` looks a bit confusing (maybe put it in a bracket) and similarly for rest of the code. Maybe using Prettier would be helpful. :D

Also, I think you can make your functions 'const'.

Also, next time onwards, please make sure to create new branches for your PRs.

**Review by anandbaburajan — CHANGES_REQUESTED** [review]
Looks good! The exporting part looks fine too! I think it would be good to throw an error or similar if a user creates two branches with the same name. Also, it would be good if the 'Sergio Flores` part is fixed too. You can create 'good-first-issues' for them too. Thanks!

File: src/graph.js [hunk 2]
```diff
@@ -39,11 +55,12 @@ class Graph extends React.Component {
...
         const branches = this.state.branches
         return (
-            <div>
+            <React.Fragment>
                 <form
                     onSubmit={(e) => {
                         e.preventDefault();
                         this.addBranch();
+
                     }}>
```
Comment (anandbaburajan) [inline src/graph.js]: Can the forms-section be a component on its own? That would be helpful to reduce the complexity of the `render()` function.

File: src/graph.js (hunk 1 above)
Comment (alwin48) [reply src/graph.js]: I will keep that in mind

File: src/graph.js (hunk 2 above)
Comment (alwin48) [reply src/graph.js]: I will try to make it a separate component

**Comment by alwin48** [comment]
> Looks good! The exporting part looks fine too! I think it would be good to throw an error or similar if a user creates two branches with the same name. Also, it would be good if the 'Sergio Flores` part is fixed too. You can create 'good-first-issues' for them too. Thanks!

I will do something about the error and default name

**Comment by anandbaburajan** [comment]
@alwin48 Do you need any input from my side or should I check something?

File: src/graph.js (hunk 2 above)
Comment (anandbaburajan) [reply src/graph.js]: Resolving this since @aryanbaburajan will take over this

**Comment by anandbaburajan** [comment]
Thanks! Merging!

**Pull request merged**